daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/SBuiltin.scala

```diff
@@ -1175,19 +1189,12 @@ private[lf] object SBuiltin {
     }
   }
 
-  final case class SBGuardTemplateId(
-      templateId: TypeConName
-  ) extends SBuiltin(2) {
-    override private[speedy] def execute(
-        args: util.ArrayList[SValue],
-        machine: Machine,
-    ): Unit = {
-      val coid = getSContractId(args, 0)
-      val record = getSRecord(args, 1)
-      if (record.id != templateId)
-        machine.ctrl = SEDamlException(IE.WronglyTypedContract(coid, templateId, record.id))
-      else
-        machine.returnValue = SBool(true)
+  final case object SBGuardConstTrue extends SBuiltinPure(1) {
```

Comment:
why do we need this as a builtin?

Reply:
We don't, it just seemed much more direct than using `SEAbs(1, SEValue(SBool(true)))` as the default for now.